extensions/cxf/deployment/src/main/java/io/quarkus/cxf/deployment/CxfProcessor.java

```diff
@@ -0,0 +1,136 @@
+        return new UnremovableBeanBuildItem(beanInfo -> {
+            Set<Type> types = beanInfo.getTypes();
+            for (Type t : types) {
+                if (WEBSERVICE_PRODUCER.equals(t.name())) {
+                    return true;
+                }
+            }
+            return false;
+        });
+    }
+
+    /**
+     * JAX-RS configuration.
+     */
+    CxfConfig cxfConfig;
+
+    @BuildStep
+    public void build(
+            CombinedIndexBuildItem combinedIndexBuildItem,
+            BuildProducer<FeatureBuildItem> feature,
+            BuildProducer<ServletBuildItem> servlet,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClass) throws Exception {
+        IndexView index = combinedIndexBuildItem.getIndex();
+
+        for (AnnotationInstance annotation : index.getAnnotations(WEBSERVICE_ANNOTATION)) {
+            if (annotation.target().kind() == AnnotationTarget.Kind.CLASS) {
+                reflectiveClass
+                        .produce(new ReflectiveClassBuildItem(true, true, annotation.target().asClass().name().toString()));
+            }
+        }
+
+        feature.produce(new FeatureBuildItem(FeatureBuildItem.CXF));
+
+        String mappingPath = getMappingPath(cxfConfig.path);
+        servlet.produce(ServletBuildItem.builder(JAX_WS_CXF_SERVLET, CXFQuarkusServlet.class.getName())
+                .setLoadOnStartup(1).addMapping(mappingPath).setAsyncSupported(true).build());
+        reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, CXFQuarkusServlet.class.getName()));
+
+        for (Entry<String, String> webServicesByPath : cxfConfig.endpoints.entrySet()) {
+            CXFQuarkusServlet.publish(webServicesByPath.getKey(), webServicesByPath.getValue());
+        }
+
+    }
+
+    private String getMappingPath(String path) {
+        String mappingPath;
+        if (path.endsWith("/")) {
+            mappingPath = path + "*";
+        } else {
+            mappingPath = path + "/*";
+        }
+        return mappingPath;
+    }
+
+    @BuildStep
+    public void createBeans(
+            BuildProducer<UnremovableBeanBuildItem> unremovableBeans,
+            BuildProducer<GeneratedBeanBuildItem> generatedBeans) throws Exception {
+        for (Entry<String, String> webServicesByPath : cxfConfig.endpoints.entrySet()) {
+            ClassOutput classOutput = new GeneratedBeanGizmoAdaptor(generatedBeans);
+            String webServiceName = webServicesByPath.getValue();
+            ClassCreator classCreator = ClassCreator.builder().classOutput(classOutput)
+                    .className(webServiceName + "Producer")
+                    .superClass(AbstractCxfWebServiceProducer.class)
+                    .build();
+            classCreator.addAnnotation(ApplicationScoped.class);
+
+            unremovableBeans.produce(new UnremovableBeanBuildItem(
+                    new UnremovableBeanBuildItem.BeanClassNameExclusion(webServiceName)));
```

Comment:
@Dufgui I think we have to remove AbstractCxfWebServiceProducer and  other UnremovableBeanBuildItem upper and fix that part with webServiceName + "Producer" ==> like classCreator .className(webServiceName + "Producer")

Reply:
Yep !